packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -895,6 +1080,687 @@ void Add(
   }
 }
 
+//Given array containing entries per row, produce rowptrs
+//Used by all add kernels
+template<typename RowPtrType>
+struct PrefixSumFunctor
+{
+  PrefixSumFunctor(RowPtrType& Crowcounts_, RowPtrType& Crowptrs_) : Crowcounts(Crowcounts_), Crowptrs(Crowptrs_) {}
+  //note: call this functor over the full length of Crowptrs (nrows+1)
+  KOKKOS_INLINE_FUNCTION void operator()(const size_t i, size_t& update, const bool finalPass) const
+  {
+    // exclusive scan
+    if(finalPass)
+    {
+      Crowptrs(i) = update;
+    }
+    update += Crowcounts(i);
+    if(finalPass)
+    {
+      if(i == Crowcounts.dimension_0() - 1)
+      {
+        Crowptrs(i + 1) = update;
+      }
+    }
+  }
+  RowPtrType& Crowcounts;
```

Comment:
See above note on Kokkos::View needing to be stored by value in functors, not by reference.